components/script/dom/location.rs

```diff
@@ -140,7 +140,10 @@ impl LocationMethods for Location {
     // https://html.spec.whatwg.org/multipage/#dom-location-host
     fn SetHost(&self, value: USVString) -> ErrorResult {
         self.check_same_origin_domain()?;
-        self.set_url_component(value, UrlHelper::SetHost);
+        // If copyURL's cannot-be-a-base-URL flag is set, terminate these steps.
```

Comment:
Can you add a comment "Step 4" to this, to make it easy to follow the code wrt the spec? (This applies to the other setters too.) Apart from that, looks great!